editor/src/components/inspector/inspector-common.ts

```diff
@@ -965,3 +972,37 @@ export function isFixedHugFillEqual(
       throw new Error(`Unknown type in FixedHugFill ${JSON.stringify(a.fixedHugFill)}`)
   }
 }
+
+export function predictElementSize(
+  metadata: ElementInstanceMetadataMap,
+  path: ElementPath,
+  changedProp: 'width' | 'height',
+  newValue: CSSNumber,
+): CanvasRectangle {
+  const elementMetadata = forceNotNull(
+    `found no metadata for element at path ${EP.toString(path)}`,
+    MetadataUtils.findElementByElementPath(metadata, path),
+  )
+  const boundingParentSizeForPercentCalc = forceNotNull(
+    'coordinateSystemBounds was null in metadata',
+    elementMetadata.specialSizeMeasurements.coordinateSystemBounds,
+  )
```

Comment:
maybe we should have a sibling to force not null called invariant. the `coordinateSystemBounds` should never be null. it's actually something we should fix in the type of specialSizeMeasurements. it has a bunch of nullable things that are actually only null for stub / spy metadata. but the _final_ SpecialSizeMeasurements should have a non-nullable coordinateSystemBounds.

Regarding 
> it can silently fail the test suite
we should fix the test suite if this can still happen. I know easier said than done.

> Can't this be just a simple return null or an Either with the reason the function failed?
if I turn this into a null or Either with a failure reason, then I just moved up the burden of throwing the error to the caller. 
1. But for the user it doesn't really matter who throws the error
2. the caller can then opt to make this a _silent_ error, without even an error message. that's not only -worse- for the user, but super painful to debug.

I guess you meant that this should be a proper user facing error message, like showing a toast.. I think in general there's something to that for actions that can sometimes fail for non-bug reasons. _But_ in this specific case (invariant violation) I think the right approach is throwing error (and making sure the tests fail on uncaught top level exceptions!!)